.github/workflows/main.yml

```
@@ -0,0 +1,63 @@
+name: Build, test and publish images
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  
+  cancel-previous-runs:
+    name: Cancel previous runs
+    runs-on: ubuntu-20.04
+    if: "!contains(github.event.commits[0].message, '[skip ci]') && !contains(github.event.commits[0].message, '[ci skip]')"
+    steps:
+      - uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ github.token }}
+
+  build-test-push:
+    name: Build, test and publish
+    runs-on: ubuntu-20.04
+    if: "!contains(github.event.commits[0].message, '[skip ci]') && !contains(github.event.commits[0].message, '[ci skip]')"
+    needs: cancel-previous-runs
+    strategy:
+      matrix:
+        version: [ '5.4', '5.5', '5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0' ]
+        flavor: [ '', '-apache', '-fpm' ]
+        include:
+          - version: 'latest'
+            flavor: ''
+    steps:
```

I suggest define that globally:

```suggestion
    env:
        VERSION: "${{ matrix.version }}${{ matrix.flavor }}"
    steps:
```

Then remove from each step individually. See example at https://github.com/hyperonecom/h1-images-recommended/blob/master/.github/workflows/image.yml#L38 .